src/main/java/net/dv8tion/jda/internal/entities/RoleImpl.java

```diff
@@ -365,9 +376,16 @@ public RoleImpl setName(String name)
     }
 
     @Override
+    @Deprecated
     public RoleImpl setColor(int color)
     {
-        this.color = color;
+        this.colors.primaryColor = color;
+        return this;
+    }
+
+    @Override
+    public RoleImpl setColors(DataObject colors) {
```

Comment:
Give it the `RoleColors` instance directly

Reply:
I am saying you should pass `RoleColors` and not `DataObject`, entity setters are supposed to be simple setters